activerecord/CHANGELOG.md

```diff
@@ -1,3 +1,20 @@
+* Use non-exist enum string to get unrelated record in My SQL
+  
+  When passing non existed enum in query, it return unrelated record in MySql.
+  Fixes #38687
+  
+    ```ruby
+    class User < ActiveRecord::Base
+      enum status: { active: 0, non_active: 1 }
+    end
+    User.where(status: :non_existing_status) 
+    # Above returns unrelated record.
+    # Going forward will raise an exception.
+    User.where(status: nil) 
+    # Above returns record with nil status.
+    ```
+  *Atul Kanswal *
```

Comment:
This changelog makes it sound like we're introducing a bug when you're actually fixing a bug. Can you reword it to the following:

```
* Raise error when non-existent enum used in query

This change will raise an error when a non-existent enum is passed to a query. Previously
with MySQL this would return an unrelated record. Fixes #38687.
```

And update the code example to:

 ```ruby
    class User < ActiveRecord::Base
      enum status: { active: 0, non_active: 1 }
    end
    User.where(status: :non_existing_status) 
    => ArgumentError ('not_defined' is not a valid status)
  ```

Reply:
🙏
Sure.
Special thanks for your talk [Breaking Down the Barrier: Demystifying Contributing to Rails](https://www.youtube.com/watch?v=7zoD6NZy6vY&t=3s).